File: sdk/src/main/java/com/cxense/cxensesdk/SdkInterceptor.java
package com.cxense.cxensesdk;

import androidx.annotation.NonNull;

import java.io.IOException;

import okhttp3.HttpUrl;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Interceptor for setting params for analytic.
 *
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2017-06-05).
 */
class SdkInterceptor implements Interceptor {
    private final String sdkName;
    private final String sdkVersion;

    SdkInterceptor(@NonNull String sdkName, @NonNull String sdkVersion) {
        this.sdkName = sdkName;
        this.sdkVersion = sdkVersion;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();
        HttpUrl url = request.url().newBuilder()
                .addQueryParameter("sdk", sdkName)
                .addQueryParameter("sdkp", "android")
                .addQueryParameter("sdkv", sdkVersion)
                .build();
        Request newRequest = request.newBuilder()
                .url(url)
                .build();
        return chain.proceed(newRequest);
    }
}

File: sdk/src/main/java/io/piano/android/cxense/SdkInterceptor.kt
package io.piano.android.cxense

import okhttp3.HttpUrl
import okhttp3.Interceptor
import okhttp3.Response

/**
 * Interceptor for setting params for analytic.
 *
 */
internal class SdkInterceptor(
    private val sdkName: String,
    private val sdkVersion: String,
) : Interceptor {
    override fun intercept(chain: Interceptor.Chain): Response =
        chain.proceed(
            chain.request()
                .run {
                    newBuilder()
                        .url(url.addSdkParams())
                        .build()
                }
        )

    private fun HttpUrl.addSdkParams() =
        newBuilder()
            .addQueryParameter(ARG_SDK_NAME, sdkName)
            .addQueryParameter(ARG_SDK_PLATFORM, VALUE_SDK_PLATFORM)
            .addQueryParameter(ARG_SDK_VERSION, sdkVersion)
            .build()

    companion object {
        const val ARG_SDK_NAME = "sdk"
        const val ARG_SDK_PLATFORM = "sdkp"
        const val ARG_SDK_VERSION = "sdkv"
        const val VALUE_SDK_PLATFORM = "android"
    }
}
